app/scenes/Settings/Details.js

```diff
@@ -14,137 +13,130 @@ import Input, { LabelText } from "components/Input";
 import Scene from "components/Scene";
 import ImageUpload from "./components/ImageUpload";
 import env from "env";
-
-type Props = {
-  auth: AuthStore,
-  ui: UiStore,
-};
-
-@observer
-class Details extends React.Component<Props> {
-  timeout: TimeoutID;
-  form: ?HTMLFormElement;
-
-  @observable name: string;
-  @observable subdomain: ?string;
-  @observable avatarUrl: ?string;
-
-  componentDidMount() {
-    const { team } = this.props.auth;
-    if (team) {
-      this.name = team.name;
-      this.subdomain = team.subdomain;
-    }
-  }
-
-  componentWillUnmount() {
-    clearTimeout(this.timeout);
-  }
-
-  handleSubmit = async (event: ?SyntheticEvent<>) => {
-    if (event) {
-      event.preventDefault();
-    }
-
-    try {
-      await this.props.auth.updateTeam({
-        name: this.name,
-        avatarUrl: this.avatarUrl,
-        subdomain: this.subdomain,
-      });
-      this.props.ui.showToast("Settings saved", { type: "success" });
-    } catch (err) {
-      this.props.ui.showToast(err.message, { type: "error" });
-    }
-  };
-
-  handleNameChange = (ev: SyntheticInputEvent<*>) => {
-    this.name = ev.target.value;
-  };
-
-  handleSubdomainChange = (ev: SyntheticInputEvent<*>) => {
-    this.subdomain = ev.target.value.toLowerCase();
-  };
-
-  handleAvatarUpload = (avatarUrl: string) => {
-    this.avatarUrl = avatarUrl;
-    this.handleSubmit();
-  };
-
-  handleAvatarError = (error: ?string) => {
-    this.props.ui.showToast(error || "Unable to upload new logo");
-  };
-
-  get isValid() {
-    return this.form && this.form.checkValidity();
-  }
-
-  render() {
-    const { team, isSaving } = this.props.auth;
-    if (!team) return null;
-    const avatarUrl = this.avatarUrl || team.avatarUrl;
-
-    return (
-      <Scene title="Details" icon={<TeamIcon color="currentColor" />}>
-        <Heading>Details</Heading>
-        <HelpText>
+import useCurrentTeam from "hooks/useCurrentTeam";
+import useStores from "hooks/useStores";
+
+function Details() {
+  const { auth, ui } = useStores();
+  const team = useCurrentTeam();
```

Comment:
Do we need to use the `useCurrentTeam` approach to access the team rather than `auth.team`?

Reply:
It's preferred 👍